src/main/java/com/zendesk/maxwell/schema/columndef/BigIntColumnDef.java

```diff
@@ -3,17 +3,39 @@
 import com.zendesk.maxwell.producer.MaxwellOutputConfig;
 
 import java.math.BigInteger;
+import java.util.Objects;
 
 public class BigIntColumnDef extends ColumnDef {
 	static private final BigInteger longlong_max = BigInteger.ONE.shiftLeft(64);
 
-	protected boolean signed;
+	private boolean signed;
 
-	public BigIntColumnDef(String name, String type, short pos, boolean signed) {
+	private BigIntColumnDef(String name, String type, short pos, boolean signed) {
```

Comment:
hmmm... I think it's kinda-sorta-ok to keep these public rather than force an API change (include someone using maxwell-as-JAR is creating them).  Maybe we could add a deprecation annotation suggesting that people move to the `#create` api instead?

Reply:
I thought it was generally okay to change API per https://maxwells-daemon.io/embedding/. The changes to enforce immutability also removes a bunch of setters, change visibility of public member fields to private, and changes expectations for ColumnDefs to all be implemented as immutable. Why are the constructor changes important to retain?

Reply:
I should've asked - what cases would someone be manually creating ColumnDefs instead of using `ColumnDef.build`?

Reply:
yeah I mean who knows what will be in the wild.   To your other point, if we're changing the api dramatically I guess this change is ok too.